lib/pbench/server/unpack_tarballs.py

```diff
@@ -0,0 +1,123 @@
+import glob
+from logging import Logger
+import os
+from pathlib import Path
+import tempfile
+from typing import NamedTuple
+
+from pbench.server import PbenchServerConfig
+from pbench.server.filetree import FileTree
+from pbench.server.report import Report
+from pbench.server.utils import get_tarball_md5
+
+
+class Results(NamedTuple):
+    total: int
+    success: int
+
+
+class Unpack:
+    LINKSRC = "TO-UNPACK"
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        """Unpacks Tarball in the INCOMING Directory
+
+        Args:
+            config: PbenchServerConfig configuration object
+            logger: A Pbench python Logger
+        """
+        self.config = config
+        self.logger = logger
+
+    def unpack(self) -> Results:
+        """Scans for tarballs in the TO-UNPACK subdirectories of the
+        ARCHIVE directory and unpacks them using FileTree.unpack() function.
+
+        Returns:
+            Results tuple containing the counts of Total and Successful tarballs.
+        """
+        tarlist = glob.iglob(self.config.ARCHIVE / "*" / self.LINKSRC / "*.tar.xz")
```

Comment:
Another serious bug. I just beat myself up for about 2 days before I figured this out in my modified version of the bash script; so, "don't be like Dave". 😆 

The unpacker is run with a "bucket" parameter. (Yes, there *is* a parameter) It's `small`, `medium`, or `large`, corresponding to the config variables `lowerbound` and `upperbound` in the appropriate `pbench-unpack-tarballs/${BUCKET}` section.

These will be *megabyte* values, which are compared against the *size* of each file. Only if a file's size is >= lowerbound and < upperbound does it get processed by the specific instance of the unpacker.

So you'll need to *stat* each file you `iglob` and compare the size against the properly scaled `lowerbound` and `upperbound` to decide whether to process that file.

(I blew it by leaving that out, causing the files I pushed to be processed in parallel by all three unpackers, with a resulting unpredictable mess that drove me crazy until I figured it out about 15 minutes ago.)

Look at the top of the bash script for the computation of (byte scaled) `lowerbound` and `upperbound`.

Reply:
Just FYI, @riya-17, `Path` has a [`glob()` method](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob) as well as a [`stat()` method](https://docs.python.org/3/library/pathlib.html#pathlib.Path.stat).